docs/RFCS/typing.md

```diff
@@ -357,16 +357,50 @@ a preference to the type on the right".
 
 This is different from casts, see below.
 
-The need for this type of extension was also part of both Rick and Morty.
+The need for this type of extension is also implicitly
+present/expressed in the alternate proposals Rick and Morty.
 
-### First pass: placeholder annotation assertion
+### First pass: placeholder annotations
 
-For the first pass, we walk through the expression tree and assign the placeholders
-that are direct arguments of explicit type annotations to those types.
+In the first pass we check the following:
 
-(= for each placeholder, if there is a type annotation around it,
-assign this annotation as the placeholder's type before the rest of type checking
-starts; fail if another earlier annotation provides a conflicting type.)
+- if any given placeholder appears as immediate argument of an
+  explicit annotation, then assign that type to the placeholder (and
+  reject conflicting annotations after the 1st).
+
+- otherwise (no direct annotations on a placeholder), if all
```

Comment:
I realized that these two rules conflict less than I thought they would, because if a placeholder is an immediate argument of an explicit annotation, it logically cannot be an immediate argument of all cast expressions as well.